### PR TITLE
Change Buffer back to Vault and add an Escrow called Buffer

### PR DESCRIPTION
**Pull request by sunbreak1211**

### CODE REVIEW

File: src/AllocatorVault.sol [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    // --- math ---
+
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        unchecked {
+            z = x != 0 ? ((x - 1) / y) + 1 : 0;
+        }
+    }
+
+    // --- getters ---
+    // In theory, `ilk.Art` should be equal than `urn.art` for this type of collateral, as there should only be one position per `ilk`.
```
Comment (telome) [inline src/AllocatorVault.sol]:
```suggestion
    // In theory, `ilk.Art` should be equal to `urn.art` for this type of collateral, as there should only be one position per `ilk`.
```

Comment (sunbreak1211) [reply src/AllocatorVault.sol]: Fixed!

Comment (sunbreak1211) [reply src/AllocatorVault.sol]: Actually fixed this in the `dev` branch, and now I have a conflict lol

Comment (sunbreak1211) [reply src/AllocatorVault.sol]: Ok, now fixed.

**Pull request merged**